__tests__/intergration.tests.js

```diff
@@ -139,13 +139,94 @@ describe("/api", () => {
   });
 });
 
+describe("/api/articles/:article_id/comments", () => {
+  test("POST: 201 accepts an object with username and body, returns the posted comment", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({
+        body: "testing, testing, 123!",
+        username: "butter_bridge",
+      })
+      .expect(201)
+      .then((response) => {
+        expect(response.body.comment).toMatchObject({
+          comment_id: expect.any(Number),
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+          author: expect.any(String),
+          body: expect.any(String),
+          article_id: 1,
+        });
+      });
+  });
+  // test("POST:400 should return Bad Request if invalid body field", ()=>{
+  //   return request(app)
+  //     .post("/api/articles/1/comments")
+  //     .send({
+  //       body: 7,
+  //       username: "butter_bridge",
+  //     })
+  //     .expect(400)
+  //     .then((response) => {
+
+  //     })
+
+  //})
+  test("POST:400 should return Bad Request for missing input fields", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({})
+
+      .expect(400)
+      .then((response) => {
+        expect(response.body.message).toBe("Bad Request");
+      });
+  });
+  test("POST:404 should return Not Found for invalid author field", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({
+        body: "testing, testing, 123!",
+        username: "Elusive",
+      })
+      .expect(404)
+      .then((response) => {
+        expect(response.body.message).toBe("Not Found");
+      });
+  });
+
+  test("POST:400 should return Bad Request for invalid article id input", () => {
+    return request(app)
+      .post("/api/articles/banana/comments")
+      .send({
+        body: "testing, testing, 123!",
+        username: "butter_bridge",
+      })
+      .expect(400)
+      .then((response) => {
+        expect(response.body.message).toBe("Bad Request");
+      });
+  });
+  test("POST:404 should return Not Found if valid but not existing article_id", () => {
+    return request(app)
+      .post("/api/articles/999/comments")
+      .send({
+        body: "testing, testing, 123!",
+        username: "butter_bridge",
+      })
+      .expect(404)
+      .then((response) => {
+        expect(response.body.message).toBe("Not Found");
+      });
```

Comment:
Really great job with your error handling for this endpoint. There is a lot to consider and you haven't missed anything :)